src/Fable.Transforms/Rust/Fable2Rust.fs

```diff
@@ -3834,6 +3845,14 @@ module Compiler =
                 $"{import.LocalIdent}"
 
             member _.GetAllImports() = imports.Values |> Seq.toList
+            member _.ClearAllImports ctx =
+                for importKv in imports do
+                    let depthsLeft = importKv.Value.Depths |> List.filter (fun dp -> dp > ctx.ModuleDepth)
+                    importKv.Value.Depths <- depthsLeft
+                    if importKv.Value.Depths.Length = 0 then
+                        imports.Remove(importKv.Key) |> ignore
+                        ctx.UsedNames.RootScope.Remove(importKv.Value.LocalIdent) |> ignore
+
```

Comment:
@alexswan10k Do you mind explaining what the import cleaning is supposed to do, what's the intent?

Reply:
Sure. So this is necessary because when you exit a scope, the used names were previously memorized. This means that the parent scope will try and declare new aliases which do not line up with the useages in the code (it thinks there are collisions). By clearing out any idents which have gone out of scope (declared in a submodule), this fixes the problem.